src/main/scala/lc/Main.scala

```diff
@@ -40,32 +41,35 @@ class Captcha(throttle: Int) {
   }
 
   def getCaptcha(id: Id): Array[Byte] = {
+    var image :Array[Byte] = null
+    var blob: Blob = null
   	imagePstmt.setString(1, id.id)
   	val rs: ResultSet = imagePstmt.executeQuery()
-  	rs.next()
-  	val blob = rs.getBlob("image")
-  	var image :Array[Byte] = null
+  	if(rs.next()){
+      blob = rs.getBlob("image")
+      updatePstmt.setString(1,id.id)
+      updatePstmt.executeUpdate()
```

Comment:
I think we should mark a captcha as solved when either
1. a solution is attempted (`checkAnswer` is called)
2. or when it is picked in `getChallenge` method

It would be preferable to keep the GET Media API endpoint simple.

Reply:
Also, based on where you decide to mark it, you can change the field name to `attempted` or `picked`.

Reply:
The reason I did not mark it as solved in the `getChallenge` method is because it only returns the token. The captcha media is still not seen by the user. 
Once someone requests the media, it is out there to be seen by anyone (bots & humans), this is when it is marked as solved.

Reply:
Alright, let's keep it this way for now, and discuss further when we add application support.